File: Server/src/main/java/Server/core/game/interaction/city/falador/WineOfZamorakInteraction.java
package core.game.interaction.city.falador;

import core.game.interaction.Option;
import core.game.interaction.SpecialGroundInteraction;
import core.game.interaction.SpecialGroundItems;
import core.game.node.entity.npc.NPC;
import core.game.node.entity.player.Player;
import core.game.world.map.RegionManager;

import java.util.List;

public class WineOfZamorakInteraction extends SpecialGroundInteraction {
    @Override
    public void handle(Player player, Option option){
        player.faceLocation(SpecialGroundItems.WINE_OF_ZAMORAK.getLocation());
        final List<NPC> npcs = RegionManager.getLocalNpcs(player);
        for (NPC n : npcs) {
            if (n.getId() == 188) {
                n.sendChat("Hands off zamorak's wine!");
                n.getProperties().getCombatPulse().attack(player);
            }
        }
    }
}

File: Server/src/main/kotlin/rs09/game/interaction/region/falador/WineOfZamorakInteraction.kt
package rs09.game.interaction.region.falador

import core.game.node.item.GroundItem
import core.game.world.map.RegionManager
import org.rs09.consts.Items
import rs09.game.content.global.action.PickupHandler
import rs09.game.interaction.InteractionListener

class WineOfZamorakInteraction : InteractionListener() {

    override fun defineListeners() {
        on(Items.WINE_OF_ZAMORAK_245,GROUNDITEM,"take"){player, wine ->
            if(player.location.regionId != 11574){
                PickupHandler.take(player, wine as GroundItem)
                return@on true
            }
            val npcs = RegionManager.getLocalNpcs(player)
            for (n in npcs) {
                if (n.id == 188) {
                    n.sendChat("Hands off zamorak's wine!")
                    n.properties.combatPulse.attack(player)
                }
            }
            return@on true
        }
    }
}
